include/seqan3/alphabet/concept.hpp

```diff
@@ -699,12 +698,20 @@ namespace seqan3
  * This is a customisation point (see \ref about_customisation). To specify the behaviour for your own alphabet type,
  * simply provide one of the three functions specified above.
  */
+#if SEQAN3_WORKAROUND_GCC_89953
 template <typename alph_t>
 //!\cond
     requires requires { { detail::adl_only::alphabet_size_fn<alph_t>{} }; } &&
              requires { { detail::adl_only::alphabet_size_obj<alph_t>() }; } // ICE workarounds
 //!\endcond
 inline constexpr auto alphabet_size = detail::adl_only::alphabet_size_obj<alph_t>();
+#else // ^^^ workaround / no workaround vvv
+template <typename alph_t>
+//!\cond
+    requires requires { { detail::adl_only::alphabet_size_fn<alph_t>{}() }; }
+//!\endcond
+inline constexpr auto alphabet_size = detail::adl_only::alphabet_size_fn<alph_t>{}();
+#endif // SEQAN3_WORKAROUND_GCC_89953
```

Comment:
Wondering if you could save some duplication here by putting just the `&& requires`  into a newline.
Depends on if it works and helps with understanding the code (the difference is clearer) and if we want to violate code style for this....

Reply:
Note that both lines

```cpp
alphabet_size = detail::adl_only::alphabet_size_obj<alph_t>();
```

and

```
alphabet_size = detail::adl_only::alphabet_size_fn<alph_t>{}();
```

are different